source/support.tex

```diff
@@ -3166,6 +3168,214 @@
+requirements\iref{utility.arg.requirements}.
+Lvalues of type \tcode{source_location}
+are swappable\iref{swappable.requirements}.
+
+\pnum
+All of the following conditions are true:
+\begin{itemize}
+\item \tcode{is_nothrow_move_constructible_v<source_location>}
+\item \tcode{is_nothrow_move_assignable_v<source_location>}
+\item \tcode{is_nothrow_swappable_v<source_location>}
+\end{itemize}
+\begin{note}
+The intent of \tcode{source_location} is
+to have a small size and efficient copying.
+\end{note}
+
+\pnum
+The data members \tcode{file_name_} and \tcode{function_name_}
+always each refer to an NTBS.
+
+\pnum
+The copy/move constructors and the copy/move assignment operators of
+\tcode{source_location} meet the following postconditions:
+Given two objects  \tcode{lhs} and \tcode{rhs} of type \tcode{source_location},
+where \tcode{lhs} is a copy/move result of \tcode{rhs}, and
+where \tcode{rhs_p} is a value denoting the state of \tcode{rhs}
+before the corresponding copy/move operation,
+then each of the following conditions is \tcode{true}:
+\begin{itemize}
+\item \tcode{strcmp(lhs.file_name(), rhs_p.file_name()) == 0}
+\item \tcode{strcmp(lhs.function_name(), rhs_p.function_name()) == 0}
+\item \tcode{lhs.line() == rhs_p.line()}
+\item \tcode{lhs.column() == rhs_p.column()}
+\end{itemize}
+
+\rSec2[support.srcloc.cons]{Creation}
+
+\begin{itemdecl}
+static consteval source_location current() noexcept;
+\end{itemdecl}
+\begin{itemdescr}
+\pnum
+\returns
+\begin{itemize}
+\item
+  When invoked by a function call
+  whose \grammarterm{postfix-expression} is
+  a (possibly parenthesized) \grammarterm{id-expression} naming \tcode{current},
+  returns a \tcode{source_location} with an implementation-defined value.
+  The value should be affected by \tcode{\#line}\iref{cpp.line}
+  in the same manner as for \tcode{__LINE__} and \tcode{__FILE__}.
+  The values of the exposition-only data members
+  of the returned \tcode{source_location} object
+  are indicated in \tref{support.srcloc.current}.
+
+\begin{libefftabvalue}
+  {Value of object returned by \tcode{current}}
+  {support.srcloc.current}
+\tcode{line_}    &
+  A presumed line number\iref{cpp.predefined}.
+  Line numbers are presumed to be 1-indexed;
+  however, an implementation is encouraged to use 0
+  when the line number is unknown. \\ \rowsep
+\tcode{column_}  &
+  An implementation-defined value denoting
+  some offset from the start of the line denoted by \tcode{line_}.
+  Column numbers are presumed to be 1-indexed;
+  however, an implementation is encouraged to use 0
+  when the column number is unknown. \\ \rowsep
+\tcode{file_name_} &
+  A presumed name of the current source file\iref{cpp.predefined} as an NTBS.
+  \\ \rowsep
+\tcode{function_name_} &
+  A name of the current function
+  such as in \tcode{__func__}\iref{dcl.fct.def.general} if any,
+  an empty string otherwise. \\
+\end{libefftabvalue}
+
+\item
+  Otherwise, that is, when invoked in some other way, returns a
```

Comment:
remove "that is,"

Reply:
We could even remove "when invoked in some other way", because that's redundant with "Otherwise".